src/Illuminate/Foundation/Console/StorageClearCommand.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+namespace Illuminate\Foundation\Console;
+
+use Illuminate\Console\Command;
+use Illuminate\Console\ConfirmableTrait;
+use Illuminate\Contracts\Filesystem\Factory as FilesystemFactory;
+use Illuminate\Contracts\Filesystem\Filesystem;
+use InvalidArgumentException;
+use Symfony\Component\Console\Attribute\AsCommand;
+
+#[AsCommand(name: 'storage:clear')]
+class StorageClearCommand extends Command
+{
+    use ConfirmableTrait;
+
+    /**
+     * The console command signature.
+     *
+     * @var string
+     */
+    protected $signature = 'storage:clear 
+                            {--disk=local : The storage disk to clear}
+                            {--folder= : The specific folder within the disk}
+                            {--force : Force the operation to run when in production}';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Clear a specific folder or all contents of a given storage disk (e.g., local, public, s3)';
+
+    /**
+     * Execute the console command.
+     *
+     * @return void
+     */
+    public function handle(FilesystemFactory $filesystem)
+    {
+        if (! $this->confirmToProceed()) {
+            return;
+        }
+
+        $diskName = $this->option('disk');
+        $folder = $this->option('folder');
+        $folder = $folder ? trim($folder, '/') : null;
+
+        try {
+            $disk = $filesystem->disk($diskName);
+        } catch (InvalidArgumentException $e) {
+            $availableDisks = implode(', ', array_keys(config('filesystems.disks')));
+            $this->error("Disk [{$diskName}] is not configured. Available disks: {$availableDisks}");
+
+            return;
+        }
+
+        try {
+            if ($folder) {
+                if (! $disk->exists($folder)) {
+                    $this->error("The folder [{$folder}] does not exist on the [{$diskName}] disk.");
+
+                    return;
+                }
+
+                $this->clearFolder($disk, $folder);
+                $this->line("Cleared folder [{$folder}] on disk [{$diskName}].");
+            } else {
+                $this->clearDisk($disk, $diskName);
+                $this->line("Cleared all contents on disk [{$diskName}].");
+            }
+        } catch (\Throwable $e) {
```

Comment:
Refactored the logic to make it cleaner and easier to follow. Now, if a folder is provided but doesn’t exist, it exits early with an error. If the folder exists, it gets cleared, otherwise the entire disk is cleared. 
This avoids deep nesting and keeps each action clearly separated.

```suggestion
        try {
            if ($folder &&  ! $disk->exists($folder)) {
               $this->error("The folder [{$folder}] does not exist on the [{$diskName}] disk.");

                return;
            }
            
            if ($folder) {
               $this->clearFolder($disk, $folder);
                $this->line("Cleared folder [{$folder}] on disk [{$diskName}].");
                
                return;
            }
                
            $this->clearDisk($disk, $diskName);
            $this->line("Cleared all contents on disk [{$diskName}].");
        } catch (\Throwable $e) {
```